### PR TITLE
Javabrake backlog

### PR DESCRIPTION
**Pull request by sumitjoshi1989**

### CODE REVIEW

File: src/main/java/io/airbrake/javabrake/Constant.java [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
-    public static String apmQuery = "queries-stats";
-    public static String apmQueue = "queues-stats";
+    protected static String apmRoute = "routes-stats";
+    protected static String apmRouteBreakDown = "routes-breakdowns";
+    protected static String apmQuery = "queries-stats";
+    protected static String apmQueue = "queues-stats";
+    protected static int maxRetryAttempt = 1;
+    private static Integer[] statusCode = { 404, 408, 409, 410, 500, 502, 504};
+    protected static int BACKLOG_FLUSH_PERIOD = 60;
+    protected static List<Integer> getStatusCodeCriteriaForBacklog() {
```
Comment (mmcdaris) [inline src/main/java/io/airbrake/javabrake/Constant.java:16]: This doesn't express that these are failure codes. What do you think of renaming this one to something like `failureCodeList()` instead.

**Review by mmcdaris — COMMENTED** [review]
Hi @sumitjoshi1989 left a few comments with some ideas.

File: src/main/java/io/airbrake/javabrake/BackLog.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package io.airbrake.javabrake;
+
+import java.io.IOException;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import okhttp3.Call;
+import okhttp3.Response;
+
+public class BackLog extends TimerTask {
```
[the remark this answers is not in the document]
Comment (sumitjoshi1989) [reply src/main/java/io/airbrake/javabrake/BackLog.java:10]: ok

File: src/main/java/io/airbrake/javabrake/Constant.java (hunk 1 above)
Comment (sumitjoshi1989) [reply src/main/java/io/airbrake/javabrake/Constant.java:16]: ok

File: src/main/java/io/airbrake/javabrake/Constant.java [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
-    public static String apmRoute = "routes-stats";
-    public static String apmRouteBreakDown = "routes-breakdowns";
-    public static String apmQuery = "queries-stats";
-    public static String apmQueue = "queues-stats";
+    protected static String apmRoute = "routes-stats";
+    protected static String apmRouteBreakDown = "routes-breakdowns";
+    protected static String apmQuery = "queries-stats";
+    protected static String apmQueue = "queues-stats";
+    protected static int maxRetryAttempt = 1;
+    private static Integer[] statusCode = { 404, 408, 409, 410, 500, 502, 504};
```
[the remark this answers is not in the document]
Comment (sumitjoshi1989) [reply src/main/java/io/airbrake/javabrake/Constant.java:14]: ok

File: build.gradle [hunk 4]
```diff
@@ -75,7 +75,7 @@ publishing {
 
             pom{
                 name = 'javabrake'
-                url = 'https://github.com/airbrake/javabrake'
+                url = 'https://airbrake.io'
```
[the remark this answers is not in the document]
Comment (sumitjoshi1989) [reply build.gradle:78]: https://mvnrepository.com/artifact/io.airbrake/demosplib/0.0.2

[the remark this answers is not in the document]
Comment (mmcdaris) [reply build.gradle:78]: Looks good, thanks for the clarification.

**Review by mmcdaris — APPROVED** [review]
Nice work Sumit 🎉

**Review by mmcdaris — APPROVED** [review]
Nice update for the response message @sumitjoshi1989 👍

**Pull request merged**